.github/workflows/build-and-tag-maven.yml

```diff
@@ -0,0 +1,107 @@
+name: Build and tag Maven project
+
+on:
+  workflow_call:
+    inputs:
+      property-to-update:
+        description: property to update in additon to the version of the pom file
+        type: string
+        required: false
+    outputs:
+      version:
+        description: "The version of the new tag created by this workflow"
+        value: ${{ jobs.build.outputs.version }}
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Alfresco/alfresco-build-tools/.github/actions/pre-commit@v1.5.0
```

Comment:
On second thought, not sure this part should be included in the reusable workflow (as it's conceptually independent of building and tagging)